ext.go

```diff
@@ -34,20 +34,47 @@ func RegisterExt(extID int8, value MarshalerUnmarshaler) {
 	})
 }
 
+func UnregisterExt(extID int8) {
+	UnregisterExtEncoder(extID)
+	UnregisterExtDecoder(extID)
+}
+
 func RegisterExtEncoder(
 	extID int8,
 	value interface{},
 	encoder func(enc *Encoder, v reflect.Value) ([]byte, error),
 ) {
 	typ := reflect.TypeOf(value)
 	extEncoder := makeExtEncoder(extID, typ, encoder)
+	oldTyp, ok := typeEncMap.Load(extID)
+	if ok {
+		typeEncMap.Delete(oldTyp)
+		typ := oldTyp.(reflect.Type)
+		if typ.Kind() == reflect.Ptr {
+			typeEncMap.Delete(typ.Elem())
+		}
```

Comment:
This is rather subtle and repeats a lot.

I guess you are trying to fully unregister the old type - pointer to the type and the type. So
- if we pass the pointer - we unregister both
- if we pas the value - we unregister only the value and leave the pointer

That does not sound right...

Reply:
I think unregister is the opposite behavior of register. What's your expected behavior?

Reply:
Is there some mistakes in my code?

Reply:
Or we call unregister func when we register?

Reply:
>Or we call unregister func when we register?

And unexport `unregister` functions? Let's do this if it works for you.

Reply:
Alternatively we can keep `UnregisterExt` but unexport `UnregisterExtEncoder` and `UnregisterExtDecoder`.

Reply:
Ok, I think we should keep UnregisterExt at least.